packages/react/src/components/Storage/FileUploader/Tracker/FileState.tsx

```diff
@@ -1,30 +1,55 @@
 import React from 'react';
 import { translate } from '@aws-amplify/ui';
-import { Text, Flex } from '../../../../primitives';
-import { CheckIcon, ErrorIcon } from '../Previewer/PreviewerIcons';
+import { Text, ComponentClassNames } from '../../../../primitives';
+import { IconCheck, IconError } from '../../../../primitives/Icon/internal';
 import { FileStateProps } from '../types';
+import classNames from 'classnames';
+import { classNameModifier } from '../../../../primitives/shared/utils';
 
 export const FileState = ({
   loading,
   success,
   error,
   paused,
   errorMessage,
+  percentage,
 }: FileStateProps): JSX.Element => {
-  if (loading) return <Text className="">{translate('Loading')}</Text>;
-  if (paused) return <Text className="">{translate('Paused')}</Text>;
+  if (loading)
```

Comment:
We may prefer `switch` in this case since it has better performance. [ref](https://medium.com/front-end-weekly/switch-case-if-else-or-a-lookup-map-a-study-case-de1c801d944)

Reply:
FYI, it is a switch in parent PR #2892